c2corg_ui/static/js/searchfilters.js

```diff
@@ -61,76 +52,121 @@ app.module.directive('appSearchFilters', app.searchFiltersDirective);
 
 /**
  * @param {angular.Scope} $scope Scope.
- * @param {angular.JQLite} $element Element.
- * @param {angular.Attributes} $attrs Attributes.
+ * @param {ngeo.Location} ngeoLocation ngeo Location service.
+ * @param {ngeo.Debounce} ngeoDebounce ngeo Debounce service.
  * @constructor
  * @ngInject
  * @export
  */
-app.SearchFiltersController = function($scope, $element, $attrs) {
-
+app.SearchFiltersController = function($scope, ngeoLocation, ngeoDebounce) {
 
   /**
    * @type {angular.Scope}
    * @private
    */
   this.scope_ = $scope;
 
-  // FIXME: not generic to all document types
-  // Init filters object
   /**
-   * @type {Array<string>}
-   * @export
+   * @type {ngeo.Location}
+   * @private
    */
-  this.activities = [];
-
+  this.location_ = ngeoLocation;
 
   /**
-   * @type {Array<string>}
+   * @type {Object}
    * @export
    */
-  this.waypoint_types = [];
-
+  this.filters = {};
 
   /**
-   * @type {string}
-   * @export
+   * @type {boolean}
+   * @private
    */
-  this.lang;
+  this.loading_ = true;
+
+  // Fill the filters according to the loaded URL parameters
+  var keys = this.location_.getParamKeys().filter(function(x) {
+    return app.SearchFiltersController.IGNORED_FILTERS.indexOf(x) === -1;
+  });
+  for (var i = 0, n = keys.length; i < n; i++) {
+    this.getFilterFromPermalink_(keys[i]);
+  }
 
+  // Deep watch is used here because we need to watch the list filters as well
+  // which a simple $watch or $watchCollection does not. Might cause
```

Comment:
OK, thanks for confirming.
I thought it was worth warning in case the list of criterias grows big and complex :)